test/extended/networking/egress_router_cni.go

```diff
@@ -0,0 +1,95 @@
+	IPv4MatchPattern = "IP Destinations: [80 UDP 10.100.3.0 8080 SCTP 203.0.113.26 80 8443 TCP 203.0.113.27 443]"
+	IPv6MatchPattern = "IP Destinations: [80 UDP 10:100:3::0 8080 SCTP 203:0:113::26 80 8443 TCP 203:0:113::27 443]"
+	TimeOut          = 1 * time.Minute
+	Interval         = 5 * time.Second
+)
+
+var _ = g.Describe("[sig-network][Feature:EgressRouterCNI]", func() {
+	var oc *exutil.CLI
+	oc = exutil.NewCLI(EgressRouterCNIE2E)
+
+	g.It("Should ensure ipv4 egressrouter cni resources are created", func() {
+		doEgressRouterCNI(EgressRouterCNIV4Manifest, oc, IPv4MatchPattern)
+	})
+	InOVNKubernetesContext(
+		func() {
+			g.It("Should ensure ipv6 egressrouter cni resources are created", func() {
+				doEgressRouterCNI(EgressRouterCNIV6Manifest, oc, IPv6MatchPattern)
+			})
+		},
+	)
+})
+
+func doEgressRouterCNI(manifest string, oc *exutil.CLI, matchString string) error {
+	var deployment *appsv1.Deployment
+	var podList *v1.PodList
+	f := oc.KubeFramework()
+
+	g.By("Creating an egressroutercni object")
+	egressRouterCNIYaml := exutil.FixturePath("testdata", "egress-router-cni", manifest)
+
+	g.By(fmt.Sprintf("Calling oc create -f %s", egressRouterCNIYaml))
+	err := oc.AsAdmin().Run("create").Args("-f", egressRouterCNIYaml).Execute()
+	e2e.ExpectNoError(err, "While creating egress-router-cni object")
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.By("Making sure egressroutercni deployment is created")
+	o.Eventually(func() error {
+		deployment, err = f.ClientSet.AppsV1().Deployments(f.Namespace.Name).Get(context.Background(), EgressRouterCNIDeployment, metav1.GetOptions{})
+		return err
+	}, TimeOut, Interval).Should(o.Succeed())
+
+	g.By("Checking network-attachment-definition resource is created")
+	output, err := oc.AsAdmin().Run("get").Args("network-attachment-definition", "-o=jsonpath={.items[0].metadata.name}").Output()
+	e2e.ExpectNoError(err, "While creating egress-router-cni network-attachement-definition")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(output).Should(o.ContainSubstring(EgressRouterCNINadObject))
+
+	g.By("Getting a pod from deployment in running state")
+	o.Eventually(func() error {
+		podList, err = e2edeployment.GetPodsForDeployment(f.ClientSet, deployment)
+		e2e.ExpectNoError(err, "While getting pods from egress-router-cni deployment")
+		return err
+	}, TimeOut, Interval).Should(o.Succeed())
+
+	o.Expect(podList.Items).NotTo(o.BeEmpty())
+	pod := podList.Items[0]
+	e2e.Logf("egress router cni pod %s is created\n", pod.Spec.Containers[0].Name)
+
+	node := pod.Spec.NodeName
+	expectNoError(err)
+	g.By("Checking no errors in egress-router log on the node " + node)
+	result, err := oc.AsAdmin().Run("debug").Args("node/"+node, "--", "chroot", "/host", "/bin/bash", "-c", "cat /tmp/egress-router-log").Output()
+	if err != nil {
+		e2e.Failf("couldn't cat egress-router cni log: %v", err)
+	}
+	if !strings.Contains(result, matchString) {
+		e2e.Failf("failed to run egress-router cni logs: %s", result)
```

Comment:
`e2e.Failf("egress-router cni logs at %s on node %s did not match %s", "/tmp/egress-router-log", node, matchString)`

I guess this would be better that outputing all the logs, aren't they extensive?

Reply:
the cni log file around 10-15 lines not extensive and shows all iptables rules installed so dumping it at the error won't be an issue

Reply:
Fair enough, then something like 
`e2e.Failf("egress-router cni logs at %s on node %s did not contain %s: %s", "/tmp/egress-router-log", node, matchString, results)`
could probably give more clues on why the test failes?